be/src/olap/data_dir.cpp

```diff
@@ -379,6 +381,21 @@ Status DataDir::load() {
         if (rowset_meta->is_local()) {
             rowset_meta->set_fs(local_fs);
         }
+        if (rowset_meta->has_delete_predicate()) {
+            // copy the delete sub pred v1 to check then
+            auto orig_delete_sub_pred = rowset_meta->delete_predicate().sub_predicates();
+            auto* delete_pred = rowset_meta->mutable_delete_pred_pb();
+            DeleteHandler::convert_to_sub_pred_v2(delete_pred, rowset_meta->tablet_schema());
```

Comment:
If the predicate is already delete pred v2, then you also rewrite the rowset meta. It is useless. And will cause the BE restart time very long.